File: Programming Basics with Python/Вложени цикли/sum_prime_non_prime.py

str_input = input()

sum_prime = 0
sum_not_prime = 0
while str_input != "stop":
    num = int(str_input)

    if num < 0:
        print("Number is negative.")
        str_input = input()
        continue

    count = 0
    for i in range(1, num + 1):
        if num % i == 0:
            count += 1
    if count == 2:
        sum_prime += num
    else:
        sum_not_prime += num

    str_input= input()

print(f"Sum of all prime numbers is: {sum_prime}")
print(f"Sum of all non prime numbers is: {sum_not_prime}")

File: Programming Basics with Java/Nested Loops/SumPrimeNonPrime.java
import java.util.Scanner;

public class SumPrimeNonPrime 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int sum_prime = 0;
        int sum_non_prime = 0;

        while (true)
        {
            String command = input.nextLine();
            if (command.equals("stop"))
            {
                System.out.printf("Sum of all prime numbers is: %d\n", sum_prime);
                System.out.printf("Sum of all non prime numbers is: %d", sum_non_prime);
                break;
            }

            int num = Integer.parseInt(command);
            if (num < 0)
            {
                System.out.println("Number is negative.");
            }
            else
            {
                boolean flag = false;
                for (int i = 2; i <= num / 2; ++i) 
                {
                    if (num % i == 0) 
                    {
                        flag = true;
                        break;
                    }
                }

                if (!flag)
                {
                    sum_prime += num;
                }
                else
                {
                    sum_non_prime += num;
                }
            }
        }
    }
}